modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/NodeImpl.java

```diff
@@ -1082,20 +1084,8 @@ public void onApplied( long lastAppliedLogIndex) {
             fsmCaller.addLastAppliedLogIndexListener(lnsr);
 
             fsmCaller.onCommitted(commitIdx);
-
-            try {
-                if (!externalAwaitStorageLatch) {
-                    applyCommitLatch.await();
-                }
-            } catch (InterruptedException e) {
-                LOG.error("Fail to apply committed updates.", e);
-
-                return false;
-            }
         } else {
-            if (externalAwaitStorageLatch) {
-                opts.getStorageReadyLatch().countDown();
-            }
+            logApplyComplition.complete(0L);
```

Comment:
Why it's 0? Despite the fact that there were no pending records in log, storage may have it's own checkpointed state with non-zero applied index. BTW, how do we use the revision, the one that is used to compete the future?

Reply:
> I think, it would be useful to know a last revision that was applied before node start. Don't you?
Yep, probably.